pkg/xds/secrets/ca_provider.go

```diff
@@ -31,12 +48,27 @@ func (s *meshCaProvider) Get(ctx context.Context, mesh *core_mesh.MeshResource)
 		return nil, nil, errors.New("CA backend is nil")
 	}
 
+	var cancel context.CancelFunc
+	timeout := backend.GetRootChain().GetRequestTimeout()
```

Comment:
nit: I think these can be declared inside the `if`

Reply:
@michaelbeaumont Yes, added this decl to avoid shadowing `ctx` within the `if` block, but it looks like declaring `cancel` within the `if`, the `ctx` will still be overwritten in the outer scope as desired.